pkg/cache/snapshot_test.go

```diff
@@ -28,19 +28,21 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/features"
 	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
 var snapCmpOpts = []cmp.Option{
 	cmpopts.EquateEmpty(),
 	cmpopts.IgnoreUnexported(ClusterQueue{}),
-	cmpopts.IgnoreFields(ClusterQueue{}, "RGByResource"),
-	cmpopts.IgnoreFields(Cohort{}, "Members"), // avoid recursion.
+	cmpopts.IgnoreFields(ClusterQueue{}, "RGByResource", "Generation"),
+	cmpopts.IgnoreFields(Cohort{}, "Members", "Generation"), // avoid recursion.
```

Comment:
maybe it's worth testing that the generation has the value that we expect

Reply:
we will only add this in snapshot cmpopts for preemption. this field will not be changed because we will not modified this field in snapshot.
we have add test to ensure this field not change in snapshot test